### DISCUSSION TITLE
Should there be documentation about multi-server installations?

### DISCUSSION THREAD

**Discussion by ThrawnCA** [ckan/ckan#6344] [Ideas]
It's quite feasible to split a CKAN installation over multiple servers. For example, we have an auto-scaling group of servers behind a load balancer to run CKAN, with a second auto-scaling group of servers that run the job workers, and Solr installed on a third set of servers (not so easy to auto-scale, but it's at least isolated so that it isn't competing for CPU and memory), and connecting to Redis in the cloud.

Should there be some documentation of what's needed to make this work, at least for a source install (which is intended as the most flexible approach)?

**Comment by Zharktas** [comment]
There's an old wiki page about HA installations https://github.com/ckan/ckan/wiki/CKAN-High-Availability. But typically tech team does not have resources to maintain complex documentation, even the current "this is how you can get CKAN running on ubuntu" takes an effort. There are at least 3 major cloud providers and multiple smaller ones which each do things slightly differently. If we have documentation for it, we will end up with support requests which we don't have resources to resolve.  And then you have to account the security too, as many of the users are not experts in server maintenance, even less is cloud maintenance.

**Reply by ThrawnCA** [reply]
I wasn't so much talking about specific cloud configuration, more like, "here is the minimum configuration that you need to make this piece work in isolation, have fun." Eg if you want to move the job workers to a separate machine, then they need a copy of the CKAN configuration file and source code, but don't need a web server.

**Reply by Zharktas** [reply]
I'm all for it if someone has time to write it :) wiki would probably be the place to maintain it, at least until it's good enough to be merged in the docs itself.
